### PR TITLE
Add option for skipping Git working tree check

### LINKED ISSUE #37: Script not working in monorepo: `Not a git repository: backend/.git`
We originally used the [strapi-starter-gatsby-corporate](https://github.com/strapi/strapi-starter-gatsby-corporate) template to set up our website. This sets up two folders in a single Git repo: `backend` (Strapi) and `frontend` (Gatsby). Now that I'm trying to upgrade/migrate it to v4, I'm running into the error below, as the script is looking for a `.git` folder in the `backend` folder, while we only have a `.git` folder in the root folder of our project.

Could you please add an option to ignore this check? I can make sure myself that my Git working tree is clean 😉 thanks!

```
% npx @strapi/codemods migrate:application backend
npx: installed 568 in 40.029s
error: Command failed with exit code 128: git --git-dir=backend/.git --work-tree=backend status --porcelain
fatal: not a git repository: 'backend/.git'
```

### PR DESCRIPTION
**Pull request by dennisameling**
Fixes #37. Happy to take a different approach if the team prefers that. This at least unblocked my workflow!

### CODE REVIEW

**Comment by markkaylor** [comment]
Hi @dennisameling thank you for the pull request, I will try to review it before the end of the week.

File: bin/commands/migrate.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ const { migratePlugin, migrateApiFolder, migrateDependencies, migrateApplication
 // Global utils
 const { isPathStrapiApp, logger, isCleanGitRepo, promptUser } = require('../../lib/global/utils');
 
-const migrate = async (type, path, pathForV4Plugin) => {
+const migrate = async (type, path, pathForV4Plugin, cliOptions) => {
```
Comment (markkaylor) [inline bin/commands/migrate.js:16]: I think we should change this to accept an object `{type, path, pathForV4Plugin, cliOptions}`

File: bin/cli.js [hunk 2]
```diff
@@ -42,9 +42,14 @@ program
...
+  .option(
+    "--skip-working-tree-check",
+    "Skip checking whether the Git working tree is clean. Especially useful for monorepos.",
+    undefined
+  )
   .description('Migrate a v3 Strapi application to v4')
-  .action(async (path) => {
-    await migrate('application', path);
+  .action(async (path, options) => {
+    await migrate('application', path, undefined, options);
```
Comment (markkaylor) [inline bin/cli.js:52]: See my other comment, then here: `{type: 'application', path, cliOptions: options}`

**Review by markkaylor — CHANGES_REQUESTED** [review]

File: bin/cli.js (hunk 2 above)
Comment (markkaylor) [reply bin/cli.js:52]: and you will have to make the update anywhere else `migrate` is called

**Comment by derrickmehaffy** [comment]
Since there has been no update to this PR I'm going to mark it as closed.

**Pull request closed without merging**